### PR TITLE
fix dissapearing user

### PR DESCRIPTION
**Pull request by bobobob22**

### CODE REVIEW

File: packages/roomSync/src/store/dancefloor/dancefloor.selector.ts [hunk 1]
```diff
@@ -23,3 +25,19 @@ export const isUserOnDancefloorSelector = (
...
   })
+
+export const avatarsVisibleOnDancefloor = createSelector(
+  [dancefloorSelectors.dancefloor, djSeatsSelectors.djSeats],
+  (dancefloorMap: Map<string, DancefloorAvatar>, djSeatsMap: Map<number, DjSeat>) => {
+    const visibleAvatars = new Map(dancefloorMap)
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const [key, value] of djSeatsMap.entries()) {
+      if (value.djPeerId && visibleAvatars.has(value.djPeerId)) {
```
Comment (JaHajRF) [inline packages/roomSync/src/store/dancefloor/dancefloor.selector.ts:36]: shouldn't we use userID here. Mobile users would not have peerId.

File: packages/roomSync/src/store/dancefloor/dancefloor.saga.ts [hunk 2]
```diff
@@ -39,9 +19,5 @@ export function* handleRemoveAvatarFromDancefloor(
 }
 
 export function* dancefloorSaga(roomBroadcast: RoomBroadcast): Generator {
-  yield all([
-    takeEvery(DjSeats.ON_TAKE_DJ_SEAT_MESSAGE, onTakeDjSeatHandleDancefloor, roomBroadcast),
```
Comment (JaHajRF) [inline packages/roomSync/src/store/dancefloor/dancefloor.saga.ts:43]: As you deleted action handling leaving and taking DjSeat does it work properly?
I mean in `Podium.container` we have functions that broadcasts take/leaveDjSeat. Then on message received, we dispatch those two removed actions.
I dont see a place where it is handled.

Comment (bobobob22) [reply packages/roomSync/src/store/dancefloor/dancefloor.saga.ts:43]: I think it is working now in that way:
1 we broadcast information and then dispatch it in `djSeatsActions` , for example `djSeatsActions.onTakeDjSeatMessage`
2. In djSeat reducer we update if dj took or left the dj place 
3. We use selector in which we can display if avatars should be visible on dancefloor (selector which check if avatar is dj or not and returns only avatars which are not dj)

Comment (JaHajRF) [reply packages/roomSync/src/store/dancefloor/dancefloor.saga.ts:43]: Sound good! ;)

Comment (JaHajRF) [reply packages/roomSync/src/store/dancefloor/dancefloor.saga.ts:43]: What do you think about changing the  `dancefloor` name of saga, selectors, reducer e.t.c? Something like 'usersInRoom' should be more appropriate as `dancefloor` might be misleading.

File: packages/signaling-server/src/store/configureStore.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ const rootReducer = {
   [keyNames.player]: playerReducer,
   [keyNames.djSeats]: djSeatsReducer,
   [keyNames.initState]: initStateReducer,
-  [keyNames.dancefloor]: dancefloorReducer,
+  [keyNames.dancefloor]: usersInRoomReducer,
```
Comment (JaHajRF) [inline packages/signaling-server/src/store/configureStore.ts:27]: keyNames enum should be also changed from `dancefloor` to `usersInRoom`

**Review by JaHajRF — APPROVED** [review]

File: packages/roomSync/src/store/usersInRoom/usersInRoom.selector.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+export const avatarsVisibleOnDancefloor = createSelector(
+  [usersInRoomSelectors.usersInRoom, djSeatsSelectors.djSeats],
+  (dancefloorMap: Map<string, DancefloorAvatar>, djSeatsMap: Map<number, DjSeat>) => {
+    const visibleAvatars = new Map(dancefloorMap)
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const [key, value] of djSeatsMap.entries()) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      for (const [visibleAvatarKey, visibleAvatarValue] of visibleAvatars.entries()) {
+        if (value.djPeerId && visibleAvatarValue.userId === value.userId) {
```
Comment (JaHajRF) [inline packages/roomSync/src/store/usersInRoom/usersInRoom.selector.ts:38]: Could we delete map entries based on userId as it's provided for both mobile and web users?

File: packages/roomSync/src/store/dancefloor/dancefloor.saga.ts (hunk 2 above)
Comment (bobobob22) [reply packages/roomSync/src/store/dancefloor/dancefloor.saga.ts:43]: sure, changed

File: packages/roomSync/src/store/dancefloor/dancefloor.selector.ts (hunk 1 above)
Comment (bobobob22) [reply packages/roomSync/src/store/dancefloor/dancefloor.selector.ts]: changed, now it is removed when userId is the same

File: packages/signaling-server/src/store/configureStore.ts (hunk 3 above)
Comment (bobobob22) [reply packages/signaling-server/src/store/configureStore.ts]: ok

**Pull request merged**